internal/users/users_test.go

```diff
@@ -128,6 +129,53 @@ func TestUpdateLocalGroups(t *testing.T) {
 	}
 }
 
+func TestCleanupSystemGroups(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		groupFilePath string
+
+		getentCmd []string
+
+		wantErr bool
+	}{
+		"Successfully cleans up groups": {groupFilePath: "root_in_groups.group"},
```

Comment:
Can you be a little bit more fine grained in the use case?

We can imagine: one user in a group, only users to be removed in a group, a mix of one local existing and one to remove users.
Also, another use case where there is nobody to remove.

That way, in case of failures, we will be more precise about the use case that is failing.